### ISSUE TITLE
Support partial matching

### ISSUE THREAD

**Issue by ageorgou** [oracc/oracc-rest#17]
It looks like we are only matching on full words. For example, searching for `Gil` will only return results that have exactly `Gil` in one of the searched fields.
Instead, we would like to return entries containing e.g. `Gilgamesh`. There should be an ElasticSearch option or different query type for this.

**Comment by ageorgou** [comment]
Have been working on this by changing the `type` of the query to `phrase_prefix`. This does return partial matches, but interprets the whole query string as a single phrase, rather than different options. For example, searching for "war ca" will return entries matching "war cat", but not "warrior cat" (before this, when using the default type, we would get entries matching "war" and/or "ca", but not "warrior" nor "cat").

Unfortunately, it looks like we can't directly have the prefix behaviour applied to each word separately.

To get the desired behaviour (entries matching one or more of "war", "warrior", "ca", "cat"), we need to run separate subqueries for each search term and combine them. Each subquery should be a multi_match query of type phrase_prefix, as above. We have two options for combining them:
- Use a [bool query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html), marking the subqueries as `should` (so that we take the union of the results):
```json
{
  "query": {
    "bool": {
        "should": [
            {
                "multi_match": {
                    "query": "cyrus",
                    "fields": ["cf", "gw"],
                    "type": "phrase_prefix"
                }
            },
            {
                "multi_match": {
                    "query": "cylinder",
                    "fields": ["cf", "gw"],
                    "type": "phrase_prefix"
                }
            }
        ]
    }
  }
}
```

- Use a [dis max query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-dis-max-query.html). As far as I can tell, this does the same thing but computes the score of results differently:

```json
{
  "query": {
    "dis_max": {
        "queries": [
            {
                "multi_match": {
                    "query": "cyrus",
                    "fields": ["cf", "gw"],
                    "type": "phrase_prefix"
                }
            },
            {
                "multi_match": {
                    "query": "cylinder",
                    "fields": ["cf", "gw"],
                    "type": "phrase_prefix"
                }
            }
        ]
    }
  }
}
```

**Comment by ageorgou** [comment]
Fixed by #20.
